### PR TITLE
Better warning behavior wöhler analyzer

### PR DESCRIPTION
**Pull request by old2rng**
Update warning behaviour for Wöhler curve analyzer methods #66 
Hint: Change of some warnings possible do to further issues #101

### CODE REVIEW

**Review by johannes-mueller — CHANGES_REQUESTED** [review]
Thanks. See my remarks about the code below.

File: src/pylife/materialdata/woehler/maxlike.py [hunk 1]
```diff
@@ -82,6 +100,8 @@ def __max_likelihood_full(self, initial_wcurve, fixed_prms):
         def warn_and_fix_if_no_runouts():
             nonlocal fixed_prms
             if self._fd.num_runouts == 0:
+                #check this estimation. Workaround could be, send error and only allow evaluation with runout...
+                # confirm with AMP4-Lab (do they evaluate maxli without any runouts)
```
[the remark this answers is not in the document]
Comment (old2rng) [reply src/pylife/materialdata/woehler/maxlike.py:104]: Thanks for your hint. I will check the comment and update code if necessary.

File: src/pylife/materialdata/woehler/maxlike.py [hunk 2]
```diff
@@ -43,6 +43,24 @@ def _specific_analysis(self, wc):
...
+        self._infinite_runout_loads = np.unique(self._infinite_runouts.load.values)
+        self._infinite_mixed_loads = np.intersect1d(self._infinite_runout_loads, self._infinite_fractured_loads)
+
+        def fail_if_less_than_two_mixed_levels():
+            if len(self._infinite_mixed_loads) < 2:
+                raise ValueError("MaxLikeHood: need at least two mixed load levels.")
+
+        def fail_if_less_than_three_fractures():
+            if self._infinite_fractures.shape[0] < 3 or len(self._infinite_fractured_loads) < 2:
+                raise ValueError("MaxLikeHood: need at least three fractures on two load levels.")
```
[the remark this answers is not in the document]
Comment (old2rng) [reply src/pylife/materialdata/woehler/maxlike.py:59]: Yes, you are right. They don't need to be classed. Thanks for your suggestion!

**Review by johannes-mueller — APPROVED** [review]
LGTM

**Pull request merged**